src/content/en/2021/caching.md

```diff
@@ -17,4 +19,146 @@ featured_stat_3: TODO
+Caching is a technique that enables the reuse of previously downloaded content. Over the past decade...
+
+
+## CDN Cache adoption
+
+Talk about CDN adoptions, 14% of desktop pages use a Cloudflare CDN.
+
+{{ figure_markup(
+  image="top-cdns.png",
+  caption="Adoption of the top CDNs.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=2035524639&format=interactive",
+  sheets_gid="58739923",
+  sql_file="top_cdns.sql"
+) }}
+
+## Service Worker adoption
+
+Adoption of Service Workers has seen a significant increase from 2020. While 1.2% of Desktop pages registered a service worker, 9% of Mobile pages in the top 1,000 registered a Service Worker.
+
+{{ figure_markup(
+  image="sw-adoption.png",
+  caption="Service Worker adoption.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=688062455&format=interactive",
+  sheets_gid="802609299",
+  sql_file="service_worker_rank.sql"
+) }}
+
+## Headers adoption
+
+Usage of the Cache-Control header has increased steadily since 2019. 74% of responses on Mobile pages included the Cache-Control header, while 75% of responses on Desktop pages utilized the header. From 2020, the usage of this specific header increased by 0.71% for Mobile and by 1.13% for Desktop.
+
+Cache-Control: DT 75%, Mobile 74%
+Expires: DT 55%, Mobile 56%
+ETag: DT 48%, Mobile 47%
+Last-Modified: DT 73%, Mobile 71%
+
+### Cache Control
+
+When using the Cache-Control header, usage increased by 1% for Desktop and remained the same for Mobile. However, usage of the Last-Modified header decreased by 1% across both interfaces.
+
+{{ figure_markup(
+  image="cache-control-adoption.png",
+  caption="The percent of requests that use the Cache-Control response header.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=885292598&format=interactive",
+  sheets_gid="2102749619",
+  sql_file="header_trends.sql"
+) }}
+
+Cache Control directives, 61% of mobile requests include a Cache-Control response header with a `max-age` directive.
+
+The most misused Cache Control directive is `set-cookie`, used for 0.07% of total directives for Desktop and 0.08% for Mobile.
+
+Fun fact the the largest max-age value is 1625540701883604800000 seconds or 51545557517871.8 years (51 trillion).
+
+{{ figure_markup(
+  caption="TODO",
+  content="51 trillion years",
+  classes="big-number",
+  sheets_gid="529870849",
+  sql_file="todo.sql"
+) }}
+
+Talk about `immutable` directive, 12% of desktop requests containing Cache-Control: immutable are on the "static.parastorage.com" host.
+
+Talk about invalid date, last modified and expires, (0.11% of dates in desktop requests are invalid).
+
```

Comment:
Desktop - Valid date strings: 
99.18% using date, 72.57% using last-modified, 55.48% using expires

Mobile - Valid date strings: 
99.07% using date, 70.55% using last-modified, 55.57% using expires

Although 99% of both Desktop and Mobile pages use valid date strings, only 55% of Desktop and 56% of Mobile pages use 'expires'. Last-modified is used at a higher percentage, at 73% for Desktop and 71% for Mobile. While the amount of invalid 'date' and 'last-modified' directives is less than 1%, 2.75% of Mobile and 3.20% of Desktop 'expires' are being used incorrectly.

What is or would be the impact from using expires more effectively? Could we possibly see big wins from getting more consistent here?